packages/bolt-connection/src/bolt/response-handler.js

```diff
@@ -196,6 +193,15 @@ export default class ResponseHandler {
   _resetFailure () {
     this._currentFailure = null
   }
+
+  _handleErrorPayload (payload) {
+    const standardizedCode = _standardizeCode(payload.code)
+    const cause = payload.cause != null ? this._handleErrorPayload(payload.cause) : undefined
+    const error = newError(payload.message, standardizedCode, cause, payload.gql_status, payload.description, payload.diagnostic_record)
+    return this._observer.onErrorApplyTransformation(
+      error
+    )
+  }
```

Comment:
The code will be not present in the cause. I suggest to make a small in the Neo4jError, extract a `GqlError` object which will be used for the causes and `Neo4jError` extends it adding the code. What do you think? This will make clear the cause don't have code.